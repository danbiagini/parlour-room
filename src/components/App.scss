.App {
  margin-top: 15px;
}

.App .app-logo {
    margin: auto;
    // height: 150px;
}

.Home .App-header {
  text-align: center;
  padding-top: 50px;
}

.Home .lander {
    padding: 40px 0;
    text-align: center;
}

.Home .lander h1 {
    font-family: "Open Sans", sans-serif;
    font-weight: 600;
    text-align: center;
}

.Home .lander p {
    color: #999;
}

select.form-control,
textarea.form-control,
input.form-control {
  font-size: 16px;
}
input[type=file] {
  width: 100%;
}

.NotFound {
  padding-top: 100px;
  text-align: center;
}

@media all and (min-width: 480px) {
  .Login {
    padding: 60px 0;
    margin: auto;
    width: 50%;
    // border: 3px solid green;
    text-align: center;
  }

  .Login form {
    margin: 0 auto;
    max-width: 320px;
  }
}
