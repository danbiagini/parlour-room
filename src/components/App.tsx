import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { NavBarUI } from "./NavBarUI";
import { AppFooter } from "./Footer";
import { store } from "../store/index";
import Routes from "./Routes";
import { ApolloProvider, ApolloClient, InMemoryCache } from "@apollo/client";
import { Box, Grommet, grommet } from "grommet";
import { deepMerge } from "grommet/utils";

const theme = deepMerge(grommet, {
  global: {
    font: {
      family: "Roboto",
      size: "18px",
      height: "20px",
    },
    colors: {
      brand: "#228BE6",
    },
  },
  formField: {
    border: {
      color: "border",
      position: "outer",
      side: "all",
    },
  },
});

const App: React.FC = () => {
  const client = new ApolloClient({
    uri: "/graphql",
    cache: new InMemoryCache(),
  });
  return (
    <Grommet theme={theme} themeMode="light">
      <Box fill flex>
        <BrowserRouter>
          <ApolloProvider client={client}>
            <Provider store={store}>
              <NavBarUI />
              <Routes />
              <AppFooter />
            </Provider>
          </ApolloProvider>
        </BrowserRouter>
      </Box>
    </Grommet>
  );
};

export default App;
