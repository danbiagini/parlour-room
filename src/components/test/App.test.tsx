import React from "react";
import App from "../App";
import TestRenderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

// Apollo uses fetch which doesn't exist in node.js
const realFetch = window.fetch;

beforeAll(() => {
  // Apollo uses fetch which doesn't exist in node.js
  window.fetch = jest.fn();
});

afterAll(() => {
  window.fetch = realFetch;
});

test("App can render correctly", () => {
  // Test first render and componentDidMount
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

describe("render the App in the DOM, and navigate", () => {
  test("render the App in the DOM, and display About component", () => {
    const { getByText } = render(<App />);
    expect(getByText("A Place for", { exact: false }));
    expect(getByText("Login"));
  });

  test("navigate to the login page", () => {
    const { getByText } = render(<App />);
    fireEvent.click(getByText("Login"));
    expect(getByText("Sign In"));
  });

  test("navigate to the signup page", () => {
    const { getByText } = render(<App />);
    fireEvent.click(getByText("Sign Up"));
    expect(getByText("Get started by signing in with Google"));
  });
});
