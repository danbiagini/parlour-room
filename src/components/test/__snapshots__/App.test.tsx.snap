// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App can render correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bELdlv"
>
  <div
    className="StyledBox-sc-13pk1d4-0 cpUkfU"
  >
    <header
      className="StyledBox-sc-13pk1d4-0 kDJoJx"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <button
        className="StyledButton-sc-323bzc-0 hMRUyI"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 dpozuV"
        >
          <img
            className="StyledImage-ey4zx9-0 jjepGE"
            height="110px"
            src="test-file-stub"
          />
          <h3
            className="StyledHeading-sc-1rdh4aw-0 gHStfv"
          >
            Pandemic Parlour
          </h3>
        </div>
      </button>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
      />
      <nav
        className="StyledBox-sc-13pk1d4-0 jIfxqx"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 bIdaiR"
        >
          <button
            className="StyledButton-sc-323bzc-0 gQBDdm"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Sign Up
          </button>
          <div
            className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
          />
          <button
            className="StyledButton-sc-323bzc-0 gQBDdm"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Login
          </button>
        </div>
      </nav>
    </header>
    <main
      className="StyledBox-sc-13pk1d4-0 kvXgc"
    >
      <h3
        className="StyledHeading-sc-1rdh4aw-0 gHStfv"
      >
        A Place for families and friends to stay connected
      </h3>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
      />
      <h4
        className="StyledHeading-sc-1rdh4aw-0 cJdkwo"
      >
        Why Pandemic Parlour?
      </h4>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
      />
      <p
        className="StyledParagraph-tbetod-0 gfTpGC"
      >
        I decided to build Pandemic Parlour for a couple reasons. I've always had a desire to create and build using technology while also providing something useful to others. When the COVID pandemic and quarantining began I noticed a few challenges with keeping connected with families and friends. What we needed was way to have the "inner parlour" for families and friends to congregate, especially for younger kids who don‘t sit still long enough for a video chat. Pandemic Parlour is a place to chat, play games and generally interact. It is designed to be kid safe, and ultimately a way to bridge the technology gap between family members of all ages.
      </p>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
      />
      <h4
        className="StyledHeading-sc-1rdh4aw-0 cJdkwo"
      >
        What's a Parlour?
      </h4>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
      />
      <p
        className="StyledParagraph-tbetod-0 gfTpGC"
      >
        A 
        <b>
          parlour
        </b>
         (or 
        <b>
          parlor
        </b>
        ) is a reception room or public space. In Medieval Christian Europe, the "outer parlour" was the room where the monks or nuns conducted business with those outside the monastery and the "inner parlour" was used for necessary conversation between resident members. In the English-speaking world of the 18th and 19th century, having a parlour room was evidence of social status.
      </p>
    </main>
    <footer
      className="StyledBox-sc-13pk1d4-0 ftYcBg"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 bIdaiR"
      >
        <a
          className="StyledAnchor-sc-1rp7lwl-0 hHCiZt"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          <div
            className="StyledBox-sc-13pk1d4-0 iZrXsK"
          >
            <img
              className="StyledImage-ey4zx9-0 jBVMKL"
              src="test-file-stub"
            />
            <div
              className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
            />
            <span
              className="StyledText-sc-1sadyjn-0 laLPYD"
            >
              Home
            </span>
          </div>
        </a>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 dIPrTC"
        >
          <a
            className="StyledAnchor-sc-1rp7lwl-0 hHCiZt"
            href="/privacy"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <span
              className="StyledText-sc-1sadyjn-0 laLPYD"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
      />
      <span
        className="StyledText-sc-1sadyjn-0 btJJDp"
      >
        © 2020 Copyright Three Bees
      </span>
    </footer>
  </div>
</div>
`;
